WorkloadManagementSystem/DB/PilotsLoggingDB.py

```diff
@@ -0,0 +1,268 @@
+""" PilotsLoggingDB class is a front-end to the Pilots Logging Database.
+    This database keeps track of all the submitted grid pilot jobs.
+    It also registers the mapping of the DIRAC jobs to the pilot
+    agents.
+
+    Available methods are:
+
+    addPilotsLogging()
+    getPilotsLogging()
+    deletePilotsLoggin()
+    addPilotsUUID()
+    setPilotsUUIDtoIDMapping()
+    addPilotsUUIDtoIDmapping()
+    
+"""
+
+__RCSID__ = "$Id$"
+
+from DIRAC  import gLogger, S_OK, S_ERROR
+from DIRAC.ConfigurationSystem.Client.Utilities  import getDBParameters
+from DIRAC.Core.Utilities.SiteCEMapping import getSiteForCE, getCESiteMapping
+import DIRAC.Core.Utilities.Time as Time
+from DIRAC.Core.DISET.RPCClient import RPCClient
+from DIRAC.ConfigurationSystem.Client.Helpers.Registry import getUsernameForDN, getDNForUsername
+from types import IntType, LongType, ListType
+import threading
+
+from sqlalchemy.sql.schema import ForeignKey, PrimaryKeyConstraint
+from sqlalchemy.sql.sqltypes import DateTime
+from sqlalchemy.orm import session, sessionmaker, scoped_session, mapper
+from sqlalchemy.engine.reflection import Inspector
+from sqlalchemy import create_engine, Table, Column, MetaData, Integer, String
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy.orm import relationship
+from sqlalchemy.exc import SQLAlchemyError
+
+import datetime
+import time
+
+DEBUG = 1
+
+metadata = MetaData()
+Base = declarative_base()
+
+#############################################################################
+class PilotsLoggingDB(  ):
```

Comment:
Only classes that make actual object-relational mapping inherit from Base, this one is just a set of methods to operate on mapped objects.